protocol/test-sequencing.md

```diff
@@ -0,0 +1,216 @@
+[Flashblocks utilizes a sidecar design](https://github.com/ethereum-optimism/design-docs/pull/86),
+called rollup-boost, that enables flash-blocks to be flexibly attached between the services.
+
+### Interop experimentation
+
+With some communication between the build processes, we can coordinate transaction-inclusion across chains.
+E.g.:
+- Host multiple builders in the same Go process.
+- Connect multiple instances with streams.
+
+Following a stream (the "shred" draft in older interop specs),
+we can signal when events happen, and when they are rolled back (callstack revert, or tx not included).
+The other builder can then use the assumed events, or even await them, to build the block of the other chain.
+
+In the extreme, this allows synchronous EVM calls between chains, without interop spec changes.
+
+With the remote-state block-building, and the tx-pool access without running a full node,
+this could potentially also serve as a replica-node, following the "horizontal tx-pool" design.
+
+## Resource Usage
+
+For testing, this separates experimental code from the production sequencing,
+to not affect resource-usage or related processing timing constraints.
+
+By encapsulating the sequencer work, we can monitor and scale it independent of regular verifier work.
+
+Long-term this can open up more scalable block-building for interop.
+
+## Single Point of Failure and Multi Client Considerations
+
+If we decide to productionize this service after test-tooling, we can support the same set of admin RPCs,
+allowing sequencer-rotation to work the same as it does today.
+
+# Alternatives Considered
+
+## Extending op-geth
+
+The block-building described here conflict with that within the `miner` package of geth:
+- the `miner` package block-building recommits blocks.
+- the `miner` package pulls in transactions from the pool eagerly.
+- the `miner` package is a difficult diff to upstream geth, something we do not want to expand.
+
+## Extending op-wheel
+
+The `op-wheel` service set up as a CLI tool, not as a service.
+`op-wheel` has a range of functions that can be deprecated.
+Expanding the tool now takes on debt of the tool, and makes it more difficult to deprecate parts.
+
+## Extending op-node
+
+The op-node only has access to the building of the initial block template,
+and the start/stop timing of block building jobs.
+It does not have tx-pool access, ability to order transactions, or EVM or state access.
+The sequencer code-path in op-node is a critical-path, and must not break due to test changes.
+
+## As sidecar between op-node and op-geth
+
+Inserting this as sidecar between op-node and op-geth is possible, but adds extra complexity, especially w.r.t. reorgs.
+See exploration in the "L1 mode" section about this.
+
+# Risks & Uncertainties
+
+Slow remote-state access can limit the usefulness in non-test settings.
+When the node is close, this may not be a problem.
```

Comment:
Any extra latency in world state access is a real killer for performance. I _think_ the besu team has finally worked out something that's viable to provide shared state storage for Infura but it took years of research and quite possibly a database redesign.

I'd suggest we should find a way to test if remote state is a show stopper for performance before doing any extra work because we think this service will be useful for production.

Reply:
Agreed, it's listed as risk for a reason. I think we can learn a lot from the test-scope first though.